### PR TITLE
feat: sales channel module

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- sales channel module implementation

### CODE REVIEW

File: packages/sales-channel/src/services/sales-channel-module.ts
Comment (olivermrbl) [inline packages/sales-channel/src/services/sales-channel-module.ts:105]: question: Should this be `@InjectTransactionManager("baseRepository_")`? cc @adrien2p

File: packages/sales-channel/src/services/sales-channel-module.ts
Comment (olivermrbl) [inline packages/sales-channel/src/services/sales-channel-module.ts]: Same as other question/

File: packages/sales-channel/src/models/sales-channel.ts
Comment (olivermrbl) [inline packages/sales-channel/src/models/sales-channel.ts]:
```suggestion
  deleted_at?: Date | null
```

File: packages/sales-channel/src/models/sales-channel.ts
Comment (olivermrbl) [inline packages/sales-channel/src/models/sales-channel.ts:18]: thought: Do we need the soft-deletable decorator here?

File: packages/sales-channel/src/models/sales-channel.ts
Comment (olivermrbl) [inline packages/sales-channel/src/models/sales-channel.ts]: suggestion: Would default to false
```suggestion
  @Property({ columnType: "boolean" })
  is_disabled? = false
```

File: packages/sales-channel/src/models/sales-channel.ts
Comment (olivermrbl) [inline packages/sales-channel/src/models/sales-channel.ts]: suggestion: Make optional
```suggestion
  @Property({ columnType: "text", nullable: true })
  description?: string | null
```

File: packages/sales-channel/src/models/sales-channel.ts
Comment (olivermrbl) [inline packages/sales-channel/src/models/sales-channel.ts]: comment: If we choose to apply these schema changes, there are likely some other places that need updating too.

File: packages/sales-channel/src/loaders/connection.ts
Comment (olivermrbl) [inline packages/sales-channel/src/loaders/connection.ts]: todo: Use Modules.SALES_CHANNEL to avoid typos like this

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Good job. Have added a couple of comments, questions, and todos :)

File: packages/sales-channel/src/services/sales-channel-module.ts
Comment (fPolic) [reply packages/sales-channel/src/services/sales-channel-module.ts:105]: yes, should be tx manager...fixed

Comment (adrien2p) [reply packages/sales-channel/src/services/sales-channel-module.ts:105]: Yes it should be tx manager

File: .changeset/strange-ligers-glow.md
Comment (adrien2p) [inline .changeset/strange-ligers-glow.md]: todo: not up to date

**Review by olivermrbl — COMMENTED** [review]
Should we replace the seed and migration scripts with the generic utils from #6021?

Also, I think it makes sense to wait for at least #6035 to get merged, so we can incorporate those changes. No reason to accumulate debt here.

#6036 can wait as it is still WIP :)

**Comment by fPolic** [comment]
> Should we replace the seed and migration scripts with the generic utils from #6021?

Updated scripts and repo to use new utils. Waiting for the 6036 to add rest.

File: packages/modules-sdk/src/definitions.ts
Comment (adrien2p) [inline packages/modules-sdk/src/definitions.ts]: question: should we remove the suffix `service` as for the latest modules?

File: packages/sales-channel/src/models/sales-channel.ts
Comment (adrien2p) [inline packages/sales-channel/src/models/sales-channel.ts]: todo: if we have a default initializer, should we also add a default at the db level? and in the config in that case?

File: packages/sales-channel/src/models/sales-channel.ts
Comment (adrien2p) [inline packages/sales-channel/src/models/sales-channel.ts:53]: todo: missing on init

File: packages/sales-channel/src/services/sales-channel-module.ts
Comment (adrien2p) [inline packages/sales-channel/src/services/sales-channel-module.ts:55]: question: should we allow one or multiple data? in that case some changes would be required.

cc @olivermrbl @riqwan some modules are providing single or multiple data. should we align on the interfaces?

File: packages/sales-channel/src/services/sales-channel-module.ts
Comment (adrien2p) [inline packages/sales-channel/src/services/sales-channel-module.ts:73]: suggestion: In some modules we are calling list with some relations and others we are serializing the return of the create. should we align on the convention?

cc @olivermrbl

File: packages/types/src/sales-channel/common.ts
Comment (adrien2p) [inline packages/types/src/sales-channel/common.ts:1]: comment: This file, should we split the interfaces into read and mutations?

cc @olivermrbl

File: packages/types/src/sales-channel/service.ts
Comment (adrien2p) [inline packages/types/src/sales-channel/service.ts]: todo: extend IModuleService

File: packages/types/src/sales-channel/service.ts
Comment (adrien2p) [inline packages/types/src/sales-channel/service.ts]: todo: rm

File: packages/types/src/sales-channel/service.ts
Comment (adrien2p) [inline packages/types/src/sales-channel/service.ts]: todo: soft/restore are missing?

File: packages/sales-channel/src/services/sales-channel-module.ts
Comment (riqwan) [reply packages/sales-channel/src/services/sales-channel-module.ts:55]: sorry, I don't follow. 

I generally support internal services following multiple data approach, and to a certain extent recommend it on entry point services, but wouldn't enforce that on the entry point services.

Comment (adrien2p) [reply packages/sales-channel/src/services/sales-channel-module.ts:55]: oups my bad, I thought it was the main service 😂

Comment (adrien2p) [reply packages/sales-channel/src/services/sales-channel-module.ts:55]: oh, but it is right?

Comment (adrien2p) [reply packages/sales-channel/src/services/sales-channel-module.ts:55]: for the public facing API it can make sense to simplify the API for single actions no?

Comment (olivermrbl) [reply packages/sales-channel/src/services/sales-channel-module.ts:55]: Think we should always support single data and multiple data in the module service i.e. the public-facing service.

E.g.
```ts
async create(data: SomeDTO, sharedContext?: Context: Promise<SomeOtherDTO>
async create(data: SomeDTO[], sharedContext?: Context: Promise<SomeOtherDTO[]>
```

And then, in some cases, a third overload using ID also makes sense e.g. in the cart module, I added:
```ts
async addLineItems(data: SomeDTO, sharedContext?: Context: Promise<CartDTO>
async addLineItems(data: SomeDTO[], sharedContext?: Context: Promise<CartDTO[]>
async addLineItems(cartId: string, data: SomeDTO[], sharedContext?: Context: Promise<CartDTO>
```

There are likely other types of overloads that also make sense, but this will differ from module to module.

As a convention, I think we should always support the first-mentioned example for almost all mutational operations in a module service.

Comment (adrien2p) [reply packages/sales-channel/src/services/sales-channel-module.ts:55]: agreed 👍

File: packages/types/src/sales-channel/common.ts
Comment (olivermrbl) [reply packages/types/src/sales-channel/common.ts:1]: Yes, so basically, take all mutational DTOs and move them to a separate file `mutations.ts` :)

**Comment by olivermrbl** [comment]
@fPolic, can I get you to address the requested changes so we can get it merged?

File: .changeset/flat-bees-laugh.md
Comment (adrien2p) [inline .changeset/flat-bees-laugh.md:2]: todo: update this file

File: packages/sales-channel/src/models/sales-channel.ts
Comment (adrien2p) [inline packages/sales-channel/src/models/sales-channel.ts]: todo: optional

File: packages/types/src/sales-channel/mutations.ts
Comment (adrien2p) [inline packages/types/src/sales-channel/mutations.ts]: todo: this file, I would privilege interfaces

**Review by adrien2p — CHANGES_REQUESTED** [review]

File: packages/sales-channel/src/models/sales-channel.ts
Comment (adrien2p) [inline packages/sales-channel/src/models/sales-channel.ts]: todo: apply new conventions everywhere

File: packages/sales-channel/src/loaders/container.ts
Comment (adrien2p) [inline packages/sales-channel/src/loaders/container.ts]: todo: you can migrate this file

File: packages/sales-channel/src/repositories/sales-channel.ts
Comment (adrien2p) [inline packages/sales-channel/src/repositories/sales-channel.ts]: todo: you can remove this file and create a type in the module types file. you can refer to my last merged pr

**Comment by fPolic** [comment]
@olivermrbl I implemented the changes, should be ready for merge now

**Review by olivermrbl — APPROVED** [review]
LGTM – pending @adrien2p's approval

**Comment by olivermrbl** [comment]
Should we merge this?

**Pull request merged**
